src/core/Object3D.js

```diff
@@ -662,6 +668,20 @@ Object.assign( Object3D.prototype, EventDispatcher.prototype, {
 
 		}
 
+		// SkinnedMesh specific
+
+		if ( this.skeleton !== undefined ) {
```

Comment:
I think `meta` initialization and `isRootObject` related codes are also necessary in addition to `extractFromCache`.
I don't really like make duplicated codes in `SkinnedMesh`.
So IMO, as @jbaicoianu mentioned, we need restructuring if we move this into `SkinnedMesh`.